### PR TITLE
Add wallabag support

### PR DESCRIPTION
**Pull request by MeneDev**
This adds support for [wallabag](https://wallabag.it/). It should work with their hosted service as well as self-hosted, but I only tested their hosted service.

To use it you need a oAuth token as described in their [documentation](https://doc.wallabag.org/en/developer/api/oauth.html).

- [x] Test
- [ ] Refactoring
- [ ] Help Document

### CODE REVIEW

**Comment by DiamondYuan** [comment]
In order to be compatible with safari in the future, please use `RequestHelper`  and add an new client in `src/common/backend/clients`.

you can refer to  `src/common/backend/clients/leanote/client.ts` and `src/common/backend/services/leanote/index.ts`

**Comment by DiamondYuan** [comment]
For better maintenance, after implementing the client, please add tests for the client.

**Comment by DiamondYuan** [comment]
I hope to write a tutorial and submit it to [webclipper/documents](https://github.com/webclipper/documents/tree/master/docs/help/account) so that users can use it better.

**Comment by MeneDev** [comment]
I also haven't found out how to validate the access_token and refresh_token. It works if you enter the correct values, but I don't understand how to add a second step for validation in the form. Do you have any pointers?

**Comment by DiamondYuan** [comment]
> I also haven't found out how to validate the access_token and refresh_token. It works if you enter the correct values, but I don't understand how to add a second step for validation in the form. Do you have any pointers?


After clicking the verify button, web clipper will call the getRepositories function.

**Comment by MeneDev** [comment]
I tried to move the implementation to it's own client and remove the axios dependency. However, when I try to add an account and click on Authentication, the console shows `Unchecked runtime.lastError: Invalid value for origin pattern undefined/*: Missing scheme separator.`, although the entered value is `https://app.wallabag.it`. I don't understand why I cannot pass this point anymore.

I hope that's still helpful for you!

File: src/common/backend/services/wallabag/form.tsx [hunk 1]
```diff
@@ -0,0 +1,104 @@
...
+  }
+  let editMode = info ? true : false;
+  return (
+    <Fragment>
+      <Form.Item
+        label={
+          <FormattedMessage id="backend.services.confluence.form.origin" defaultMessage="Origin" />
+        }
+      >
+        {form.getFieldDecorator('wallabag_host', {
```
Comment (DiamondYuan) [inline src/common/backend/services/wallabag/form.tsx:40]: field name must be `origin`

**Comment by DiamondYuan** [comment]
> I tried to move the implementation to it's own client and remove the axios dependency. However, when I try to add an account and click on Authentication, the console shows `Unchecked runtime.lastError: Invalid value for origin pattern undefined/*: Missing scheme separator.`, although the entered value is `https://app.wallabag.it`. I don't understand why I cannot pass this point anymore.
> 
> I hope that's still helpful for you!

https://github.com/webclipper/web-clipper/blob/eebc4a2ecf34fda83d5c32620fd68b1c52412f70/src/hooks/useOriginForm.tsx#L12

File: src/common/backend/services/wallabag/form.tsx (hunk 1 above)
Comment (DiamondYuan) [reply src/common/backend/services/wallabag/form.tsx:40]: pr set `originKey `
https://github.com/webclipper/web-clipper/blob/eebc4a2ecf34fda83d5c32620fd68b1c52412f70/src/hooks/useOriginForm.tsx#L12

**Comment by MeneDev** [comment]
I think I got it now. One thing that is suboptimal is that I wrap the requests with a try/catch to throw `UnauthorizedError` because the `RequestHelper` has no support for response interceptors.

In other places the equivalent solution looks like this:

```typescript
request.interceptors.response.use(
  response => {
    if (response.clone().status === 401) {
      throw new UnauthorizedError(
        localeService.format({
          id: 'backend.services.dida365.unauthorizedErrorMessage',
          defaultMessage: 'Unauthorized! Please Login Dida365 Web.',
        })
      );
    }
    return response;
  },
  { global: false }
);
```

**Pull request closed without merging**